ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/participant_pruning_service.proto

```diff
@@ -59,6 +59,10 @@ message PruneRequest {
   //   1. no application using this participant node relies on divulgence OR
   //   2. divulged contracts on which applications rely have been re-divulged after the `prune_up_to` offset.
   bool prune_all_divulged_contracts = 3;
+
+  // Must be a valid domain ID
+  // Required
+  string domain_id = 4;
```

Comment:
Specifying a domain ID makes sense if we expose everything on the basis of an individual domain. B then this means that we can't prune stuff that is not related to a particular domain.

For example, we can't prune party allocation and package upload `Update`s from the participant event log with this.

Reply:
I see... currently pruning is working over the "merged domain offset", right (which we do not plan to have atm)? I guess: Since we need to think anyway about multi domain, and packages/parties, we could tackle that together.